src/drivers/sifive_pwm0.c

```diff
@@ -0,0 +1,340 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <metal/machine/platform.h>
+
+#ifdef METAL_SIFIVE_PWM0
+#include <metal/clock.h>
+#include <metal/compiler.h>
+#include <metal/drivers/sifive_gpio0.h>
+#include <metal/drivers/sifive_pwm0.h>
+#include <metal/io.h>
+#include <metal/machine.h>
+#include <metal/time.h>
+#include <stdio.h>
+
+/* Register fields */
+#define METAL_PWMCFG_STICKY (1UL << 8)
+#define METAL_PWMCFG_ZEROCMP (1UL << 9)
+#define METAL_PWMCFG_DEGLITCH (1UL << 10)
+#define METAL_PWMCFG_ENALWAYS (1UL << 12)
+#define METAL_PWMCFG_ENONESHOT (1UL << 13)
```

Comment:
Nit, I think xxx_ALWAYS and xxx_ONESHOT is suffice

Reply:
All above mentioned macros are in use. Zero glitch avoids any glitches in output waveform, while changing duty cycle. _ZEROCMP is used to set base PWM frequency.